src/store/todoSlice.ts

```diff
@@ -0,0 +1,48 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+export interface Todo {
+  id?: number;
+  task: string;
+  isCompleted: boolean;
+}
+
+interface TodoState {
+  todos: Todo[];
+  todoItem: Todo | undefined;
+}
+
+const initialState: TodoState = {
+  todos: [],
+  todoItem: undefined,
+};
+
+export const todoSlice = createSlice({
+  name: "todoList",
+  initialState,
+  reducers: {
+    addTodo: (state, action: PayloadAction<Todo>) => {
+      state.todos.push(action.payload);
```

Comment:
you should never do mutable operations on the state. a reducer should always return a new value of state if it is changed.